### ISSUE TITLE
Audio is choppy on both speakers and hdmi, but not bluetooth [linux]

### ISSUE THREAD

**Issue by KayJay7** [moonlight-stream/moonlight-qt#997]
**Description**
When using moonlight to connect to a pc the audio is choppy, around every second it does a brief weird noise. On HDMI it's even worse, the sound is basically off, with some sporadic sound fragment or noise.
If I'm listening through bluetooth headphones there is no issue, the sound is perfect.
I thought it might be an issue with my sound card because it affected analog audio, but it affects hdmi too.
I don't think it's an issue with audio drivers because every other application (system, firefox, vlc, discord, some game) works correctly. Increasing the realtime priority and reducing niceness of the pipewire process does nothing.

**Steps to reproduce**
Just using moonlight with those audio outputs produces choppy sound.

**Other Moonlight clients**
It doesn't occur on the android client, and neither on the windows client (on a different machine).
But it does occur both with the snap and flatpak client.
Moreover, it happens both with gamestream and sunshine as server.

**Moonlight settings**
I have increased resolution and bitrate, and forced hardware decode in the video section.
None of these affect the issue, reverting to default does nothing.

**Client PC details**
 - OS: fedora 38
 - Moonlight Version: 4.3.1
 - GPU: Intel UHD 620
 - Ffmpeg version: 6.0 from rmpfusion
 - Vaifo and glxinfo reported below in additional context

**Server PC details**
 - OS: Windows 10 19045
 - GeForce Experience version: 3.27.0.112
 - Sunshine version: 0.19.1.2215ab98b401907401676dfd3b8a126a4107ed2e
 - Nvidia GPU driver: 531.68
 - Antivirus and firewall software: Windows Defender and Windows Firewall

**Moonlight Logs**
[hdmi.log](https://github.com/moonlight-stream/moonlight-qt/files/11294337/hdmi.log)
[speakers.log](https://github.com/moonlight-stream/moonlight-qt/files/11294338/speakers.log)

**Additional context**
```
Trying display: wayland
libva info: VA-API version 1.18.0
libva info: Trying to open /usr/lib64/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_18
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.18 (libva 2.18.1)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 23.1.6 ()
vainfo: Supported profile and entrypoints
      VAProfileNone                   :	VAEntrypointVideoProc
      VAProfileNone                   :	VAEntrypointStats
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Simple            :	VAEntrypointEncSlice
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointFEI
      VAProfileH264Main               :	VAEntrypointEncSliceLP
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointFEI
      VAProfileH264High               :	VAEntrypointEncSliceLP
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline:	VAEntrypointFEI
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSliceLP
      VAProfileVP8Version0_3          :	VAEntrypointVLD
      VAProfileVP8Version0_3          :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointFEI
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileHEVCMain10             :	VAEntrypointEncSlice
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointVLD
```
```
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) UHD Graphics 620 (KBL GT2) (0x5917)
    Version: 23.0.2
    Accelerated: yes
    Video memory: 11709MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) UHD Graphics 620 (KBL GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 23.0.2
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 23.0.2
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 23.0.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```
```
#
# Sound card
#
- card: 0
  id: PCH
  name: HDA Intel PCH
  longname: HDA Intel PCH at 0xdc248000 irq 160
  driver_name: HDA-Intel
  mixer_name: Realtek ALC257
  components: HDA:10ec0257,17aa2258,00100001 HDA:8086280b,80860101,00100000
  controls_count: 39
  pcm:
    - stream: PLAYBACK
      devices:
        - device: 0
          id: ALC257 Analog
          name: ALC257 Analog
          subdevices:
            - subdevice: 0
              name: subdevice #0
        - device: 3
          id: HDMI 0
          name: HDMI 0
          subdevices:
            - subdevice: 0
              name: subdevice #0
        - device: 7
          id: HDMI 1
          name: HDMI 1
          subdevices:
            - subdevice: 0
              name: subdevice #0
        - device: 8
          id: HDMI 2
          name: HDMI 2
          subdevices:
            - subdevice: 0
              name: subdevice #0
    - stream: CAPTURE
      devices:
        - device: 0
          id: ALC257 Analog
          name: ALC257 Analog
          subdevices:
            - subdevice: 0
              name: subdevice #0
alsactl: rawmidi_device_list:105: snd_ctl_rawmidi_next_device
```

**Comment by KayJay7** [comment]
I've done some testing. I thought this was related to #830, so I tried using my system's SDL 2.26 (with no results), and an [older SDL 2.0](https://koji.fedoraproject.org/koji/rpminfo?rpmID=31097375), but none of these attempts worked.

I ran an [SDL audio demo](https://github.com/aminosbh/sdl2-audio-sample) to make sure that I can actually get working audio from sdl, and it worked.

Then, I tried setting `--env=ML_AUDIO=...` to test other backends, but only SDL made any sound. With ALSA and libsoundio, I got a "failed to open audio device, audio will be unavailable in this session" error in a toast notification.

I should note that, as mentioned in other similar issues, sound is broken for the entire system while using moonlight (even if muted from a mixer). And it does not affect bluetooth audio for some reason.
But unlike those issues, I'm using pipewire, not pulseaudio.

The workaround I'm using is: "unplugging" (not just muting) moonlight from audio devices using  [helvum](https://gitlab.freedesktop.org/pipewire/helvum) and streaming the audio separately with [sonobus](https://github.com/sonosaurus/sonobus). :pleading_face:

**Comment by Maarten-buelens** [comment]
I seem to have the same problem. When a moonlight stream starts the audio of the whole system cuts out. First when i was using ubuntu it worked but when i switched to pop os this happened.

**Comment by KayJay7** [comment]
I think I found the solution. For some reason the sampling rate of pipewire was set to `19200` by defaut (which is not normal, maybe a fedora thing, not sure). I noticed it while looking for buffer size or scheduling settings as seen in other pipewire/pulse troubleshooting threads.

I set this two properties in `/etc/pipewire/pipewire.conf`:
```
context.properties = {
    default.clock.rate          = 48000
    default.clock.allowed-rates = [ 48000 44100 96000 19200 ]
    ...
```
It seems to be working fine.
